R/collate-derivative.R

```diff
@@ -6,7 +6,8 @@ if (!exists("publish_data", mode = "function")) source(here::here("R", "publish-
 
 #' collate_derivative
 #' @param derivative `CollatedDerivative` object to calculate
-collate_derivative <- function(derivative) {
+#' @param publish Bool attempt publishing?
+collate_derivative <- function(derivative, publish=TRUE) {
   datasets <- DATASETS[names(DATASETS) %in_ci% lapply(derivative$locations, function(dsl) { dsl$dataset })]
```

Comment:
Is DATASET a global here? Is there a reason not to pass in?

Reply:
It is a global constant - it should be the original list of every dataset so that it can be filtered to the datasets used by the derivative being processed. It could be passed in and the outer layer load it from the global instead.

Reply:
Makes sense. My preference is not to use globals in general unless it is really required as it makes testing the in/out of function calls hard.